### ISSUE TITLE
Bug with format and for-loop

### ISSUE THREAD

**Issue by sobolevn** [gforcada/flake8-builtins#14]
I have spotted that this example gives me false negative.
No errors is created, but there should be one.

```python
def validate_date(date: str) -> datetime:
    """
    Validates that value is a valid date.

    >>> validate_date('20.12.2010')
    datetime.datetime(2010, 12, 20, 0, 0, tzinfo=<UTC>)

    """
    valid_formats = [
        '%d.%m.%y',
        '%d.%m.%Y',
        '%e.%m.%y',
        '%e.%m.%Y',
    ]

    for format in valid_formats:  # format is built-in name, but no error
        try:
            return datetime.strptime(date, format)  # no error here either
        except ValueError:
            continue
```

`format` is a built-in name that is not checked for some reason.

**Comment by gforcada** [comment]
Thanks for the report!

The list of builtins is gathered automatically, so the bug should be probably elsewhere (sorry I'm with my phone right now).

I will double check though. 

Would you mind creating a pull request that adds this examples as tests? Thanks in advance if you add them, if not I will do it sometime later.

**Comment by sobolevn** [comment]
@gforcada PR is coming!

**Comment by gforcada** [comment]
This was merged and released already.
